File: seunggon/easy.js
// week1_easy: 폰켓몬
// https://school.programmers.co.kr/learn/courses/30/lessons/1845

function solution(nums) {
  const inputSet = new Set(nums);
  // 중복없앤 게 배열길이 절반 이상이거나 더 적거나

  if (inputSet.size >= nums.length / 2) {
    return nums.length / 2;
  } else {
    return inputSet.size;
  }
}

File: seunggon/week5/easy.kt
// lv2_피보나치수
// https://school.programmers.co.kr/learn/courses/30/lessons/12945?language=kotlin

class Solution {
    fun solution(n: Int): Int {
        val dp = IntArray(n + 1)

        dp[0] = 0
        dp[1] = 1

        for (i in 2..n) {
            dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567
        }

        return dp[n]
    }
}